src/avt/Queries/Pick/avtPickByNodeQuery.C

```diff
@@ -230,6 +235,23 @@ avtPickByNodeQuery::Execute(vtkDataSet *ds, const int dom)
     if (!pickAtts.GetFulfilled())
         return;
 
+    //
+    // There are times when we aren't able to retrieve information
+    // for a patricular variable from the database query. For example,
+    // expressions are unavailable for query. In those cases, we can
+    // try to directly query the current dataset.
```

Comment:
I'm curious, due to the need to have this new call to RetrieveVarIno, is the logic above  (which may be allowing the call to RetrieveVarInfo on line 221 to be skipped)  too restrictive?  @aowen87 did you investigate that path at all?

Reply:
Yeah, that's a good question. I think the earlier conditions for calling this method are appropriate/not too restrictive. The reason that I put this extra call here is that we won't know whether or not we actually need to make an extra call until we've performed the Query on line 240. After that, we _should_ have variable info, unless something wen't wrong. We could just make a call to RetrieveVarInfo all the time, but I think it would be performing needless work in most cases.

Reply:
I could combine this call with the one on line 272 and just set a boolean flag in the for loop. I'm not sure if that would be more or less straight forward for an incoming reader. Thoughts?

Reply:
Since you discovered the need in making a call to RetrieveVarInfo if the database query didn't find all the information, am I right in assuming in this use-case the first call to RetrieveVarInfo is never made ?  If it **_is_** called, then I think we would need to figure out why it didn't get the needed information in the first case.  Otherwise, I think the way you have coded it is fine, especially since I cannot recall the reasons why the first call may need to be skipped, and I don't want to mess up other use-cases.

Reply:
Yeah, you are correct. I double checked just to make sure.

Reply:
So now I am wondering, it was originally skipped for a reason ... was it because we wouldn't have all the needed/correct information in order to retrieve the information here.  In which case, is this extra call actually getting the correct information? From the correct location?

Reply:
Here's my take:

The first call is only made if there aren't any material selections and we didn't create ghosts. There's a note in avtDatabase::Query that says the incident elements will be incorrect if the above conditions are true, so that information is instead taken from the database.

In this new case, we should only make this call if we've determined that the incident elements need to be retrieved from the database and we were unable to retrieve other variable information. I was assuming that RetrieveVarInfo wouldn't overwrite the incident elements if they already existed, but I think I was wrong there. I should update this call to include the incident elements found from the database: `RetrieveVarInfo(ds, id, pickAtts.GetIncidentElements())`. I should probably also make sure that the 3rd call to this function isn't made if we make this second one, as they'd be doing the exact same work.

I'll make these changes and re-test.

Reply:
FWIW, this particular dataset skips the first call because ghosts were created. It's not exactly clear why that causes an initial skip, and I don't see any notes about that, but it appears safe to make the call after the database Query (regardless of ghost status). The incident elements are the same in this case, so the baselines didn't change.

Reply:
Great job ensuring the method isn't called again unnecessarily, and thanks for looking into all my questions!